src/Microsoft.ServiceFabric.Actors/Diagnostics/EventSourceDiagnosticEvents.cs

```diff
@@ -0,0 +1,189 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using Microsoft.ServiceFabric.Actors.Runtime;
+using Microsoft.ServiceFabric.Diagnostics;
+using Microsoft.ServiceFabric.Services.Remoting;
+
+namespace Microsoft.ServiceFabric.Actors.Diagnostics
+{
+    internal class EventSourceDiagnosticEvents : IDiagnosticEvents
+    {
+        readonly ServiceContext serviceContext;
+        readonly IClock clock;
+        readonly string actorType;
+        readonly ActorFrameworkEventSource eventSource;
+        readonly Dictionary<long, ActorMethodInfo> actorMethodInfo;
+
+        public EventSourceDiagnosticEvents(ActorFrameworkEventSource eventSource, IClock clock, ServiceContext serviceContext, ActorMethodFriendlyNameBuilder nameBuilder, ActorTypeInformation typeInfo)
+        {
+            this.eventSource = eventSource ?? throw new ArgumentNullException(nameof(eventSource));
+            this.serviceContext = serviceContext;
+            this.clock = clock ?? throw new ArgumentNullException(nameof(clock));
+            _ = nameBuilder ?? throw new ArgumentNullException(nameof(nameBuilder));
+            _ = typeInfo ?? throw new ArgumentNullException(nameof(typeInfo));
+            this.actorType = typeInfo.ImplementationType.ToString();
+            this.actorMethodInfo = InitializeActorMethodInfo(nameBuilder, typeInfo);
+        }
+
+        private Dictionary<long, ActorMethodInfo> InitializeActorMethodInfo(ActorMethodFriendlyNameBuilder nameBuilder, ActorTypeInformation typeInfo)
+        {
+            var actorMethodInfo = new Dictionary<long, ActorMethodInfo>();
+
+            foreach (var actorInterfaceType in typeInfo.InterfaceTypes)
+            {
+                nameBuilder.GetActorInterfaceMethodDescriptionsV2(actorInterfaceType, out var interfaceId, out var actorInterfaceMethodDescriptions);
+                foreach (var actorInterfaceMethodDescription in actorInterfaceMethodDescriptions)
+                {
+                    var methodInfo = new ActorMethodInfo(actorInterfaceMethodDescription.MethodInfo);
+
+                    actorMethodInfo[Util.GetInterfaceMethodKey((uint)interfaceId, (uint)actorInterfaceMethodDescription.Id)] = methodInfo;
+                }
+            }
+            return actorMethodInfo;
+        }
+
+        public void AcquireActorLockFailed(DiagnosticsManagerActorContext diagnosticContext)
+        {
+            // Intentionally left blank, since we don't track
+        }
+
+        public void AcquireActorLockFinish(PendingActorMethodDiagnosticData diagnosticData, DateTime startTime)
+        {
+            if (this.eventSource.IsPendingMethodCallsEventEnabled())
```

Comment:
We can omit this. without losing any readability in small methods like this. I always omit this. and when the method starts to feel like it needs it, that's a sign to break it up.